osl/sandbox.go

```diff
@@ -70,6 +70,9 @@ type Sandbox interface {
 
 	// restore sandbox
 	Restore(ifsopt map[string][]IfaceOption, routes []*types.StaticRoute, gw net.IP, gw6 net.IP) error
+
+	// ApplyOSTweaks applies operating system specific knobs on the sandbox
+	ApplyOSTweaks()
```

Comment:
It would probably be good to flesh this interface out a bit.  As written, I don't get what this function actually means.  If there are special OS tweaks that apply to all sandboxes, then why not do it in NewSandbox().  If it only applies to some sandboxes (which is what we really want here) how does the sandbox determine whether it is special?  Right now, we just have this optionally called in createLoadBalancerSandbox(), but what if there are other types of sandboxes that need OS tweaks down the road?  Do they all get the same sets of tweaks?

There are at least two two general paths forward that have better "code hygiene".
1.  Rename this to ApplyLoadBalancerOSTweaks() -- that makes it clear that this is just for tweaking that particular special type of sandbox.
2.  Change the method signature to receive some kind of parameters indicating what kinds of tweaks to apply.  A simplistic view would be a bool parameter indicating the LB sandbox status.  But a more generic approach would be a `map[string]string` parameter which indicates some kind of OS-specific setting that should be applied.   In the case of load balancer sandboxes it would likely be something as simple as `map[string]string{"loadBalancer":"true"}`.   The OS specific sandbox could would then look through the keys to see which of key's it might have a tweak perhaps modulating how to apply it based on the value.  Then it would know what to try to apply.

Option 1 is the simplest and its reasonably clean.  I would go with this if we really don't foresee any other types of sandbox tweaks down the road ever.

Option 2 may be over-engineering a bit for the purpose.  But it is the better long term interface IMHO.  If we imagine that there could be other OS-specific tweaks, then we'd have to change the sandbox interface again down the road with Option 1.  Option 2, by contrast would require changes only to the backend code for those OSes affected by the new option and nothing to the interface at all.

BTW, with Option 2 controller.go code above would then change to:
```go
sb.osSbox.ApplyOSTweaks(map[string]string{"loadBalancer": fmt.Sprintf("%v", sb.loadBalancer)})
```
Come to think of it, it would be good to options well documented so, we might want to put the key ("loadBalancer") as a constant in osl/sandbox.go.

There may well be other options.  That's just what has occurred to me so far.

Reply:
I was trying to keep it simple because I'm not sure which is going to be the next use of it, so did not want to push into a direction that makes it less flexible when the next purpose will be found.
Anyway I agree with the general POV that this is a too simplistic and not power approach.
I personally don't like the option 1, only because will put assumption on the sandbox interface, that not necessarily reflects a generic sandbox implementation.
I like more the second option, so my proposal is to change the interface of the sandbox method like:
```
ApplyOSTweaks(tweaks map[string]*kernel.OSValue)
```
WDYT?

Reply:
Option 2 sounds sounds cleaner way for long run .

Reply:
Was the idea that each OSL would declare something like:
```
var SandboxLoadBalancerTweaks = map[string]*kernel.OSValue{...}  // possibly empty
```
and then the createLoadBalancerSandbox() function would do something like:
`sb.ApplyOSTweaks(osl.SandboxLoadBalancerTweaks)`?

This could be a reasonable approach.  I think that means less complexity in the solution with the cost of requiring changes in all (ok: all 2-3 of them  :) ) OSLs whenever we add a sandbox tweak.